### PR TITLE
Items can cause damage when thrown at players again

### PR DESCRIPTION
**Pull request by MaxIsJoe**
CL: [Improvement] Players vomiting from being thrown around is now locked behind a 75% roll to avoid players from abusing the vomit stuns more frequently.
CL: [Fix] Fixed an issue where players did not receive damage upon getting hit by an object. This will make spears useable again.
CL: [Fix] Fixed an issue where the vomit emote would force the player to dry heave twice.

### CODE REVIEW

**Comment by Bod9001** [comment]
hummm I think we should just increase the speed which the vomit triggers at?

File: UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs [hunk 1]
```diff
@@ -1711,61 +1677,119 @@ public void FlyingUpdateMe()
...
+			//TODO: DamageTile( goal,Matrix.Matrix.TilemapsDamage);
+			if (hit == null) continue;
+			if (NewtonianMovement.magnitude < iav2.ThrowSpeed * 0.25f) continue;
+			var damage = (iav2.ServerThrowDamage);
+			if (hit.TryGetComponent<Integrity>(out var integrity) && isServer)
+			{
+				integrity.ApplyDamage(damage, AttackType.Melee, iav2.ServerDamageType);
+			}
+			//TODO: Catching
+			//(Max): Catching what?
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs]: In ss13 if you're holding the throw button, you can Catch items being thrown at you

File: UnityProject/Assets/Scripts/Player/MovementV2/MovementSynchronisation.cs [hunk 2]
```diff
@@ -340,7 +340,7 @@ public override void Awake()
 	public void ImpactVomit(UniversalObjectPhysics ob, Vector2 Newtonian)
 	{
 		if (VomitEmote == null) return;
-		if (Newtonian.magnitude > 6.5f)
+		if (Newtonian.magnitude > 7f && DMMath.Prob(75))
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Player/MovementV2/MovementSynchronisation.cs:343]: ok you did cool

File: UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs]: oh, you could have explained this further but sure ig.

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs]: updated the comment to reflect that

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**